src/dfx/src/lib/operations/canister/deploy_canisters.rs

```diff
@@ -99,7 +115,7 @@ pub async fn deploy_canisters(
     let pool = build_canisters(
         env,
         &canisters_to_load,
-        &canisters_to_deploy,
+        &canisters_to_build,
```

Comment:
Let me use the e2e project as an example.
The `app` canister depends on `dep_b` and `dep_c`.
When `dfx deploy app`, both `dep_b` and `dep_c` must be built so that their candid files can be copied to `.dfx/local/...` and the path of candids are required when building `app` canister.

`canisters_to_build` includes `dep_b`, `dep_c`, `app`.
`canisters_to_install` contains only `app`.